370-auth-ui/session.md

```diff
@@ -0,0 +1,171 @@
+      `accounts.example.com` cannot share session with `forum.example.com`.
+    - To share sessions across services, this attribute should be set to a
+      common suffix of the service domains. For the previous example, a
+      suitable suffix would be `example.com`.
+- Developer can configured whether the cookie is a HTTP session cookie:
+    - If it is configured as a HTTP session cookie, the cookie would be deleted
+      from client side when browser session ends.
+    - If it is configured as a HTTP permanent cookie, the cookie would be
+      persisted until the maximum lifetime of session is reached.
+
+#### Configuration
+
+Sample configuration:
+```yaml
+session:
+    lifetime: 86400
+    idle_timeout_enabled: true
+    idle_timeout: 300
+    cookie_same_site: None
+    cookie_domain: example.com
+    cookie_expiration: false
+clients:
+    - auth_api_use_cookie: false
+```
+
+- **session**: Configuration for the session mechanism.
+    - **lifetime**: The maximum session lifetime in seconds,
+                    default to 2592000 (30 days).
+    - **idle_timeout_enabled**: Whether session idle timeout is enabled,
+                                default to false.
+    - **idle_timeout**: The session idle timeout period in seconds,
+                        default to 300 (5 minutes).
+    - **cookie_same_site**: The `SameSite` attribute of cookie,
+                            default to `Lax`, and other possible values are
+                            `None` and `Strict`.
+    - **cookie_domain**: The `Domain` attribute of cookie, optional.
+    - **cookie_expiration**: Whether the session cookie would expire
+                             (i.e. HTTP permanent cookie), default to false.
+- **clients**: Configuration of OIDC clients. Details are in other specs.
+    - **auth_api_use_cookie**: Whether the Auth API would set session cookie,
+                               instead of returning OIDC access/refresh tokens,
+                               default to false.
+
+### Session Management
+
+#### List
+
+Skygear Auth provides an API to list all valid sessions and offline grants of
+the user. The list includes the session attributes.
+
+### Session Resolution
+
+Skygear Auth provides an API to resolve user info from cookie / OIDC access
+token.
+
+For cookies, a session can be directly resolved from the session cookie.
+
+For OIDC access token, its parent grant would be resolved:
+- if it is a access grant: the associated session would be resolved.
+- if it is a offline grant: the offline grant itself would be resolved.
+
+After resolving a valid session/offline grant, some attributes of it would be
+updated according to the request:
+- last access time
+- last IP
+- last access user agent
+- extra information
+
+### Session Invalidation
+
+#### Implicit Invalidation
+
+A session can be implicitly invalidated if some states of the
+user changed:
+- The identity of session is deleted.
+- The user is disabled.
+
+#### Revocation
```

Comment:
Maybe we should also implement the [OAuth token revocation endpoint](https://tools.ietf.org/html/rfc7009)

cc @carmenlau